### ISSUE TITLE
False positive for class definition in function scope

### ISSUE THREAD

**Issue by hauntsaninja** [PyCQA/pylint#4590]
### Steps to reproduce

```
λ cat zzz.py
def f():
    class X(A): ...
    print(x)

class A: ...
x = 5

λ pylint zzz.py -d R,C
************* Module zzz
zzz.py:2:12: E0601: Using variable 'A' before assignment (used-before-assignment)
zzz.py:2:4: W0612: Unused variable 'X' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 1.43/10 (previous run: 1.43/10, +0.00)
```

### Expected behavior

No error

### pylint --version output

```
pylint 2.8.2
astroid 2.5.6
Python 3.7.7 (default, Apr 11 2021, 16:51:31) 
[Clang 11.0.3 (clang-1103.0.32.62)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce this.
